### PR TITLE
Zq/update conv2d by aclnn

### PR DESCRIPTION
**Pull request by NeosZhang**
## Motivation and Context

update conv2d by aclnn

## Description

update conv2d by aclnn
image

## Checklist
**Before PR**:

- [x] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: impl/ascend/functions/conv2d.cpp [hunk 1]
```diff
@@ -4,126 +4,113 @@
...
     }
-    runner.run();
+    diopiSize_t outputPadding{outputPaddingData, 2};
+
+    int64_t strideExpandData[2];
+    int64_t paddingExpandData[2];
+    int64_t dilationExpandData[2];
+
+    strideExpandData[0] = stride.data[0];
+    strideExpandData[1] = (stride.len == 1) ? stride.data[0] : stride.data[1];
```
Comment (yangbofun) [inline impl/ascend/functions/conv2d.cpp:30]: 这里是不是前面要DIOPI_CHECK下 stride, padding, dilation 的0<len <=2

File: impl/ascend/functions/conv2d.cpp [hunk 2]
```diff
@@ -4,126 +4,113 @@
...
+        outputPaddingData[i] = 0;
+    }
+    diopiSize_t outputPadding{outputPaddingData, 2};
+
+    int64_t strideExpandData[2];
+    int64_t paddingExpandData[2];
+    int64_t dilationExpandData[2];
+
+    strideExpandData[0] = stride.data[0];
+    strideExpandData[1] = (stride.len == 1) ? stride.data[0] : stride.data[1];
```
Comment (yangbofun) [inline impl/ascend/functions/conv2d.cpp:83]: 同上

File: impl/ascend/functions/conv2d.cpp (hunk 1 above)
Comment (NeosZhang) [reply impl/ascend/functions/conv2d.cpp:30]: conv2d的输入的stride, padding, dilation 正确性需要在diopi层检查吗
我理解这个是上层确认的事情。

Comment (yangbofun) [reply impl/ascend/functions/conv2d.cpp:30]: > conv2d的输入的stride, padding, dilation 正确性需要在diopi层检查吗？ 我理解这个是上层确认的事情。

理论上对任何进入diopi上的数都要做检查。

**Review by yangbofun — APPROVED** [review]

**Pull request merged**